libr/debug/p/native/linux/linux_debug.c

```diff
@@ -509,3 +514,1397 @@ RList *linux_desc_list (int pid) {
 	closedir (dd);
 	return ret;
 }
+
+
+/* Coredump  functions */
+
+static map_file_t mapping_file = {0,0};
+
+static char *prpsinfo_get_fname(FILE *f) {
+        char *p;
+        int c;
+        int pos;
+
+        p = malloc(16);
+        pos = 0;
+        while ((c = fgetc(f)) != EOF && c != '\0')
+                p[pos++] = c;
+
+        if (c == '\0')
+                p[pos] = c;
+
+        return p;
+}
+
+static char *get_basename(char *pfname, int len) {
+        char *p;
+
+        for (p = pfname + len; p != pfname ; p--)
+                if (*p == '/')
+                        return (p + 1);
+
+        return p;
+}
+
+static char *prpsinfo_get_psargs(FILE *f, char *pfname, int size_psargs) {
+        char paux[80];
+        char *p;
+        int c;
+        int pos;
+        int read;
+
+        p = malloc (size_psargs);
+
+        strcpy (p, pfname);
```

Comment:
Considering `p` was allocated on a size different from `strlen(pfname)` it may make sense to use a `strncpy` or something safer to ensure we do not write out of bounds.